### DISCUSSION TITLE
안녕하세요! 멀티플레이 서버 접속시 리스폰 위치에 대한 질문입니다.

### DISCUSSION THREAD

**Discussion by kimyang-sun** [naverz/zepeto-studio-kor#1146] [5. Q&A]
안녕하세요!
멀티플레이 서버 접속할때 리스폰 위치에 대해 궁금한 점이 있습니다.

이전 글들을 검색해보니까 OnJoinPlayer에 있는 position 값을 지정해줌으로써 접속시 위치를 변경할 수 있다고 하여
참고하여 수정하였습니다.

하지만 그렇게 하니까 

![image](https://user-images.githubusercontent.com/56675004/183570893-39ef4051-d9ed-4a16-b887-48ca80b4fb13.png)

forEach로 반복문이 돌아가면서 OnJoinPlayer가 실행될 때
새롭게 접속하는 인원에게 이미 접속한 사람들의 위치가 지정해준 위치로 보이게 되는 문제점이 발생했습니다.

새로 접속하는 사람의 위치만 -0.99, 0, 65.3 으로 하고싶은데
이미 접속해있던 다른 플레이어의 위치마저도 -0.99, 0, 65.3 으로 보이게 되는 상태입니다 ㅠ.ㅠ

![image](https://user-images.githubusercontent.com/56675004/183571012-6b13dead-5710-4b4d-a8be-ca197d1cae4c.png)

사진에서처럼 저렇게 분류해서 if 문이나 아니면 다른 방법으로 새로 입장하는 플레이어만 예외처리를 할 방법이 있을까요??
계속 시도해보다가 도저히 떠오르지가 않아서 질문 올려봅니다  :(

**Comment by ZdevMonZ** [comment]
OnStateChange에서 isFirst로 최초 접속을 판별할수 있어요 
https://studio.zepeto.me/guides/multiplay-room-state

**Reply by kimyang-sun** [reply]
@ZdevMonZ  안녕하세요! 답변 감사합니다.

말씀해주신대로 ifFisrt를 이용해서 

![image](https://user-images.githubusercontent.com/56675004/183812365-574f20f3-c07f-4a21-a6e3-8dd069fd302c.png)

OnStateChange 안에서 OnJoinPlayer가 실행되는 부분에 isFirst를 파라미터로 넣어서 

![image](https://user-images.githubusercontent.com/56675004/183816298-d0937728-60d4-4525-a49c-a8e99db12729.png)

이렇게 if 문으로 해봤는데 이미 접속해 있는 플레이어에게는 콘솔이 

![image](https://user-images.githubusercontent.com/56675004/183812917-04f27fef-ea5b-40b3-9168-2687fd90968c.png)

이렇게 뜨면서 새로운 플레이어가 나타나질 않고,

새로 접속하는 플레이어에게는 

![image](https://user-images.githubusercontent.com/56675004/183812979-a8311e49-f1ae-4bd9-8c4d-911b4c76ef8e.png)

기존에 접속하고 있는 플레이어까지도 이런식으로 isFirst 가 true로 적용이 되어서 해당 지점으로 캐릭터가 위치해 있네요...

이런식으로 사용하는게 맞는 방법일까요???

**Comment by smlee2952** [comment]
저는 index.ts - onJoin()에서 초기 위치를 잡아주고, ClientStarter.ts - OnJoinPlayer()에서 player 위치 값을 설정하도록 하였습니다. 아래 코드를 참고 하세요!

![image](https://user-images.githubusercontent.com/49216499/184079019-8b2380e1-cf82-4193-9d04-36f61bfdb47d.png)

![image](https://user-images.githubusercontent.com/49216499/184079103-341c7a1f-8431-493e-b06a-9bc8c72cea5d.png)

**Reply by zlmrzlo** [reply]
이 방식이 맞는 것 같아요. 저도 이런 식으로 했었습니다.

**Reply by kimyang-sun** [reply]
오 다들 답변 너무 감사드려요! 참고해서 적용하도록 하겠습니다!

**Reply by chlguswls32** [reply]
리스폰 위치 잘변경되셧나요? 저는 위 방법으로 했는데, 리스폰위치 변경이 되지않아서 질문드립니다.

**Reply by kimyang-sun** [reply]
@chlguswls32 
네네 저는 이 방법으로 하니까 잘 작동됐습니다.

**Reply by chlguswls32** [reply]
그렇군요..! 저는 왜안되는지 좀더확인해봐야겠습니다. 답변 감사합니다.

**Reply by zlmrzlo** [reply]
@chlguswls32 UnityEngine 라이브러리랑 스키마랑 혼동되어 사용되면 Vector3나 Transform이 vsCode나 vsStudio에서 잡아내지 못하는 것 같아요. 그 부분도 한 번 확인해보실래요?

**Reply by chlguswls32** [reply]
넵 확인해보겠습니다 ㅎㅎ 고마워요
